### ISSUE TITLE
index.d.ts is not match the API

### ISSUE THREAD

**Issue by Diluka** [baryon/tracer#85]
```ts
export const logger: AppLogger = tracer.colorConsole({ // this is line 44
    format: [
        "[{{timestamp}}] [{{title}}] [{{path}}:{{line}}] [{{processTag}}] [{{message}}]",
        {
            error: "[{{timestamp}}] [{{title}}] [{{path}}:{{line}}] [{{processTag}}] [{{message}}] \n{{stack}}"
        }
    ],
    dateformat: "yyyy-mm-dd HH:MM:ss.l",
    preprocess: function (data) {
        data.path = data.path.replace(config.APPLICATION_CONTEXT_PATH, "");
        data.title = data.title.toUpperCase();
        data.processTag = processTag;
    },
    level: config.tracer.level,
    transport: function (data) {
        if (loggerWriteStream !== process.stdout) {
            process.stdout.write(data.output + "\n");
        }
        loggerWriteStream.write(data.output + "\n");
    }
});
```

```
Error:(44, 54) TS2345: Argument of type '{ format: (string | { error: string; })[]; dateformat: string; preprocess: (data: LogOutput) => v...' is not assignable to parameter of type 'LoggerConfig'.
  Types of property 'format' are incompatible.
    Type '(string | { error: string; })[]' is not assignable to type 'string'.
```
```
Error:(55, 14) TS2339: Property 'processTag' does not exist on type 'LogOutput'.
```
```
Error:(60, 39) TS2339: Property 'output' does not exist on type 'string'.
```
```
Error:(62, 38) TS2339: Property 'output' does not exist on type 'string'.
```

**Comment by baryon** [comment]
any miss? could you give a patch?

**Comment by myfjdthink** [comment]
some problem，this is PR  https://github.com/baryon/tracer/pulls

**Comment by baryon** [comment]
fixed in v0.9.1

transport function parameter data type is LogOutput not string #87
 Merged

**Comment by Diluka** [comment]
It's still there. Here is the type error in `v0.9.1` using typescript `2.9.2`

@baryon @myfjdthink 

```
Error:(44, 54) TS2345: Argument of type '{ format: (string | { error: string; })[]; dateformat: string; preprocess: (data: LogOutput) => v...' is not assignable to parameter of type 'LoggerConfig'.
  Types of property 'format' are incompatible.
    Type '(string | { error: string; })[]' is not assignable to type 'string'.
```

```
Error:(55, 14) TS2339: Property 'processTag' does not exist on type 'LogOutput'.
```
